src/utils.rs

```diff
@@ -44,11 +40,15 @@ pub fn rewind(start: &Node, closed: &Vec<Node>) -> Vec<Coords2D> {
 }
 
 pub fn direction(current: Coords2D, parent: Coords2D) -> (i32, i32) {
-    //Calculate direction and restrict to range -1..=1
-    let mut direction_x = current.0 as i32 - parent.0 as i32;
-    let mut direction_y = current.1 as i32 - parent.1 as i32;
-    direction_x = max(-1, min(direction_x, 1));
-    direction_y = max(-1, min(direction_y, 1));
+    (
+        current.0.cmp(&parent.0) as i32,
```

Comment:
This might seem weird, but it's actually quite cool. I initially had thought of doing:
```rs
if current.x < parent.x {
  -1
} else if current.x > parent.x {
  1
} else {
  0
}
```
And then repeat for the y, but I found it a bit non-Rusty. Then I though of using std::cmp::PartialEq, which returns an enum called `Ordering`, which would end up as:
```rs
let dir x = match current.0.cmp(&parent.0) {
    Ordering::Lesser => -1,
    Ordering::Greater => 1,
    Ordering::Equal => 0,
};
```
But then I looked at the source code of `Ordering`:
```rs
#[derive(Clone, Copy, PartialEq, Debug, Hash)]
#[stable(feature = "rust1", since = "1.0.0")]
pub enum Ordering {
    /// An ordering where a compared value is less than another.
    #[stable(feature = "rust1", since = "1.0.0")]
    Less = -1,
    /// An ordering where a compared value is equal to another.
    #[stable(feature = "rust1", since = "1.0.0")]
    Equal = 0,
    /// An ordering where a compared value is greater than another.
    #[stable(feature = "rust1", since = "1.0.0")]
    Greater = 1,
}
```
See those `= -1`, `= 0`, `= 1`? That means you've got the exact values you need! And thanks to Rust's magic, you can cast an enum into an integer will get you those values directly as the exact primitive you need.

Reply:
That is very cool. I would never have thought of using ordering for that. I'd initially used the same
```
if current.x < parent.x {
  -1
} else if current.x > parent.x {
  1
} else {
  0
}
```
as yourself, but it felt a little too verbose. Casting Ord into an int is perfect.